### PR TITLE
Allow render card without display title

### PR DESCRIPTION
**Pull request by taitus**
## References

Requirement:
> c. Making the titles on cards not mandatory

## Objectives
Allow the card to be rendered without displaying the title, but still be accessible to screen readers.

## Visual Changes

- Admin cards form: Render new display_title field
Render new display title field on admin cards form

- Admin cards index: Render value for new display_title field
Render new field value on admin cards index

- Render card without title but is accessible to screen readers
Title is not visible when render card

### CODE REVIEW

**Comment by taitus** [comment]
@decembrya Can you send us the link to https://primariabrasovenilor.ro/ where we can see the example Olivia showed us at the last meeting? We remember cards with a “.” instead of a title, but we can’t find it.

**Comment by decembrya** [comment]
Hi @taitus, the page on https://primariabrasovenilor.ro/  that Olivia showed is nolonger up, but we've set something similar up in our test environment: https://buget.heroesof.tech/despre-bugetare-participativa. See the bottom two cards

File: app/components/custom/admin/widget/cards/row_component.html.erb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<tr id="<%= dom_id(card) %>" class="homepage-card">
+  <td>
+    <% if card.label.present? %>
+      <%= card.label %><br>
+    <% end %>
```
Comment (javierm) [inline app/components/custom/admin/widget/cards/row_component.html.erb]: Is this change related to this pull request? :thinking:

File: app/components/custom/admin/widget/cards/row_component.html.erb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    <%= card.title %>
+  </td>
+  <td><%= card.display_title? ? t("admin.shared.true_value") : t("admin.shared.false_value") %></td>
```
Comment (javierm) [inline app/components/custom/admin/widget/cards/row_component.html.erb]: We usually use `shared.yes` and `shared.no`. It looks like the `true_value` and `false_value` were accidentally added a few years ago :relieved:.

File: config/locales/custom/admin.yml [hunk 3]
```diff
@@ -196,3 +197,10 @@ ro:
         create: Creează o nouă înregistrare de date locale
       new:
         creating: Creează o nouă înregistrare de date locale
+
+en:
+  admin:
+    site_customization:
+      pages:
+        cards:
+          display_title_help_text: "If it is not visible, it shall still be accessible to persons using screen readers"
```
Comment (javierm) [inline config/locales/custom/admin.yml]:
```suggestion
          display_title_help_text: "If it is not visible, it will still be accessible to people using screen readers"
```

File: spec/system/custom/admin/widgets/cards_spec.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
+require "rails_helper"
+
+describe "Cards", :admin do
+  scenario "Create with title only visible for screen readers" do
+    visit admin_homepage_path
+    click_link "Create card"
+
+    expect(find("#widget_card_display_title")).to be_checked
+    expect(page).to have_field("Display the title?", checked: true)
```
Comment (javierm) [inline spec/system/custom/admin/widgets/cards_spec.rb]: The first line seems redundant :thinking:.

```suggestion
    expect(page).to have_field("Display the title?", checked: true)
```

File: spec/system/custom/admin/widgets/cards_spec.rb [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    click_button "Create card"
+
+    expect(page).to have_content "Card created successfully!"
+    expect(page).to have_css(".homepage-card", count: 1)
+
+    within "#cards" do
+      within all("tbody tr").last do
+        expect(page).to have_content "Card text"
+        expect(page).to have_content "Card description"
+        expect(page).to have_content "Yes"
```
Comment (javierm) [inline spec/system/custom/admin/widgets/cards_spec.rb]: I'm not following :thinking:. We're unchecking the "Display the title?" option, but then we're checking that "Yes" appears on the table; we might have a bug here :thinking:.

**Review by javierm — COMMENTED** [review]
@taitus Here's the review; let me know what you think!

File: config/locales/custom/activerecord.yml [hunk 6]
```diff
@@ -26,4 +28,6 @@ en:
 
     attributes:
       budget/investment:
-        heading_id: "Category"
\ No newline at end of file
+        heading_id: "Category"
+      widget/card:
+        display_title: "Display the title?"
```
Comment (javierm) [inline config/locales/custom/activerecord.yml]: Usually we don't add question marks to checkboxes :thinking:.

File: app/views/custom/admin/widget/cards/_form.html.erb [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+  <div class="row card-form-translations">
+    <%= f.translatable_fields do |translations_form| %>
+      <div class="small-12 medium-6 column end">
+        <%= translations_form.text_field :label %>
+      </div>
+
+      <div class="column">
+        <%= translations_form.text_field :title %>
+      </div>
```
Comment (javierm) [inline app/views/custom/admin/widget/cards/_form.html.erb]:
```suggestion
      <div class="small-12 medium-6 column end">
        <%= translations_form.text_field :title %>
      </div>

      <div class="column">
        <%= translations_form.text_field :label %>
      </div>
```

So now the title is closer to the "display title" checkbox.

File: config/locales/custom/activerecord.yml [hunk 8]
```diff
@@ -16,6 +16,8 @@ ro:
         main_link_url: "Linkul vă duce la (adăugați un link)"
       budget/phase/translation:
         main_link_text: "Textul de pe link"
+      widget/card:
+        display_title: "Afișați titlul?"
```
Comment (javierm) [inline config/locales/custom/activerecord.yml]: Usually we don't add question marks to checkboxes :thinking:.

**Review by javierm — COMMENTED** [review]
@taitus Looking good! I've mentioned a couple of minor things I've noticed.

**Review by javierm — APPROVED** [review]

**Comment by javierm** [comment]
@decembrya As usual, we've added a couple of texts in Romanian ([a label](https://github.com/code4romania/consul/pull/44/files#diff-693e522a555db173e5cf81b448791f7cbe305586aa220984eff2c9fcc9333da4R20) and [a hint](https://github.com/code4romania/consul/pull/44/files#diff-11dbd4692727f44bb9253ebaea66feec9587cfcb70264f74bd879b0b4d418918R190) for the new field). Could you review them? Thanks! :pray:

File: config/locales/custom/admin.yml [hunk 9]
```diff
@@ -191,6 +191,7 @@ ro:
           create: Creează o pagină nouă
         cards:
           create_card: Creează card
+          display_title_help_text: "În cazul în care nu este vizibilă, aceasta trebuie să fie totuși accesibilă persoanelor care utilizează cititoare de ecran"
```
Comment (decembrya) [inline config/locales/custom/admin.yml]: Small change here
"În cazul în care nu este vizibil, acesta trebuie să fie totuși accesibil persoanelor care utilizează cititoare de ecran"

**Comment by javierm** [comment]
The test failure is not related to this pull request; it failed on this version of Consul sometimes.
